### PR TITLE
Add historical staking balances

### PR DESCRIPTION
**Pull request by deangalvin-cb**
### What changed? Why?
This adds Historical Staking Balances

### CODE REVIEW

**Comment by cb-heimdall** [comment]
<h1>🟡 Heimdall Review Status</h1>
<table>
<tr>
<th>Requirement</th>
<th>Status</th>
<th>More Info</th>
</tr>
<tr>
<td>Reviews</td>
<td>
🟡
<code>0/1</code>
</td>
<td>
<details>
<summary>Denominator calculation</summary>
<table>
<tr>
<td>
<details>
<summary>Show calculation</summary>
<table>
<tr>
<td><code>1</code> if user is bot</td>
<td>0</td>
</tr>
<tr>
<td><code>1</code> if user is external</td>
<td>0</td>
</tr>
<tr>
<td>
From <code>.codeflow.yml</code>
</td>
<td>1</td>
</tr>
<tr>
<td>
Additional review requirements
<details>
<summary>Show calculation</summary>
<table>
<tr>
<td><strong>Max</strong></td>
<td><strong>0</strong></td>
</tr>
</table>
</details>
</td>
<td>0</td>
</tr>
<tr>
<td>
From <code>CODEOWNERS</code>
</td>
<td>0</td>
</tr>
<tr>
<td>Global minimum</td>
<td>0</td>
</tr>
<tr>
<td><strong>Max</strong></td>
<td>
<strong>
1
</strong>
</td>
</tr>
</table>
</details>
</td>
<td>1</td>
</tr>
<tr>
<td><code>1</code> if commit is unverified</td>
<td>0</td>
</tr>
<tr>
<td><strong>Sum</strong></td>
<td><strong>1</strong></td>
</tr>
</table>
</details>
</td>
</tr>
</table>

File: pkg/coinbase/stakeable_balances.go [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+}
+
+// FetchStakingBalances fetches the staking balances for a given address and asset.
+func (c *Client) fetchStakingBalances(ctx context.Context, assetId string, address *Address, o ...StakingBalanceOption) (*stakingBalances, error) {
+	req := client.GetStakingContextRequest{
+		NetworkId: address.NetworkID(),
+		AssetId:   assetId,
+		AddressId: address.ID(),
+		Options: map[string]string{
+			"mode": "default",
```
Comment (drohit-cb) [inline pkg/coinbase/stakeable_balances.go:71]: nit: can use the const defined in utils.go

File: pkg/coinbase/staking_balance.go
Comment (drohit-cb) [inline pkg/coinbase/staking_balance.go:1]: can this be renamed to `historical_balance` similar to nodejs

File: pkg/coinbase/stakeable_balances.go
Comment (drohit-cb) [inline pkg/coinbase/stakeable_balances.go:1]: can this be renamed to `staking_balance.go` similar to nodejs

File: pkg/coinbase/staking_balance.go
Comment (drohit-cb) [reply pkg/coinbase/staking_balance.go:1]: and also the core struct type

File: pkg/coinbase/stakeable_balances.go
Comment (drohit-cb) [reply pkg/coinbase/stakeable_balances.go:1]: ok `staking_balance` is related to validator balances. Maybe this for now can be `staking_context_balances.go` ?

Comment (deangalvin-cb) [reply pkg/coinbase/stakeable_balances.go:1]: love it!

File: pkg/coinbase/staking_balance.go
Comment (drohit-cb) [reply pkg/coinbase/staking_balance.go:1]: ok nvm - realized after DMs with you that historical_balance is something else within the SDK.

File: pkg/coinbase/staking_context_balances.go [hunk 2]
```diff
@@ -0,0 +1,106 @@
+package coinbase
+
+import (
+	"context"
+
+	"github.com/coinbase/coinbase-sdk-go/gen/client"
+)
+
+// stakingBalances represents the active stakeable balances for a given address and asset.
+type stakingBalances struct {
```
Comment (drohit-cb) [inline pkg/coinbase/staking_context_balances.go:10]: Did we wanna rename the type as well to something like `StakingContextBalances` ? currently very similar to `StakingBalance`

Comment (deangalvin-cb) [reply pkg/coinbase/staking_context_balances.go:10]: Yeah thats a good idea!

**Review by drohit-cb — APPROVED** [review]

**Pull request merged**